src/middlewared/middlewared/utils/io.py

```diff
@@ -8,7 +8,7 @@ def write_if_changed(path, data):
 
     changed = False
 
-    with open(os.open(path, os.O_CREAT | os.O_RDWR), 'wb+') as f:
+    with open(os.open(path, os.O_CREAT | os.O_RDWR, 0o644), 'rb+') as f:
```

Comment:
Not all files are 644, we have mode specified in the plugin for some files, don't we?

Reply:
I honestly can't see why any of the config files for the various services would need to be executable. Only possibility I can think of is if we're using the `etc` plugin to generate scripts that are run by various services which is quite plausible I guess. Nonetheless, you're right. We have the `mode` key in `etc` plugin. I don't really feel like testing changing the default mode for all consumers of this method so I'll change in `etc` plugin.